### PR TITLE
Removed the model overload from createMachine

### PR DESCRIPTION
**Pull request by mattpocock**

### CODE REVIEW

File: packages/core/src/Machine.ts [hunk 1]
```diff
@@ -69,7 +55,7 @@ export function createMachine<
   // Ensure that only the first overload matches models, and prevent
   // accidental inference of the model as the `TContext` (which leads to cryptic errors)
   config: TContext extends Model<any, any, any, any>
-    ? never
+    ? 'Passing typeof model to the first generic of createMachine is no longer supported - use model.createMachine instead'
```
Comment (mattpocock) [inline packages/core/src/Machine.ts:58]: @Andarist will this work as intended?

Comment (mattpocock) [reply packages/core/src/Machine.ts:58]: Also, there are now only two overloads left in this file. They look very similar in signature - can we condense them into... only... ONE OVERLOAD...?

Comment (mattpocock) [reply packages/core/src/Machine.ts:58]:
```ts
export function createMachine<
  TContext,
  TEvent extends EventObject = AnyEventObject,
  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
>(
  // Ensure that only the first overload matches models, and prevent
  // accidental inference of the model as the `TContext` (which leads to cryptic errors)
  config: TContext extends Model<any, any, any, any>
    ? 'Passing typeof model to the first generic of createMachine is no longer supported - use model.createMachine instead'
    : MachineConfig<TContext, any, TEvent>,
  options?: Partial<MachineOptions<TContext, TEvent>>
): StateMachine<TContext, any, TEvent, TTypestate>;
export function createMachine<
  TContext,
  TEvent extends EventObject = AnyEventObject,
  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
>(
  config: MachineConfig<TContext, any, TEvent>,
  options?: Partial<MachineOptions<TContext, TEvent>>
): StateMachine<TContext, any, TEvent, TTypestate> {
  return new StateNode<TContext, any, TEvent, TTypestate>(
    config,
    options
  ) as StateMachine<TContext, any, TEvent, TTypestate>;
}
```

File: .changeset/weak-zebras-pull.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+"xstate": minor
+---
+
+Removed the ability to pass a model as a generic to `createMachine`, in favour of `model.createMachine`. This lets us cut an overload from the definition of `createMachine`, meaning errors become more targeted and less cryptic.
```
Comment (Andarist) [inline .changeset/weak-zebras-pull.md:5]: This was already being implemented in https://github.com/statelyai/xstate/pull/2518/files#diff-00f777398251992f42d7061fc6a79e0c6e7c8aebfd257cb1d8c04129c84e0273R51 and accepted back then so I think it's an OK change.

File: packages/core/src/Machine.ts [hunk 3]
```diff
@@ -69,7 +55,7 @@ export function createMachine<
   // Ensure that only the first overload matches models, and prevent
```
Comment (Andarist) [inline packages/core/src/Machine.ts:55]: this comment should be adjusted now (or even removed)

File: packages/core/src/Machine.ts (hunk 1 above)
Comment (Andarist) [reply packages/core/src/Machine.ts:58]: Note that `createMachine` has only a single overload here now - the second thing is like a private declaration. If possible you could still consolidate both though

**Comment by Andarist** [comment]
@mattpocock this test should be removed now:
https://github.com/statelyai/xstate/blob/b0799ac8e68cabe36eff5ad61be3f0aec9314a24/packages/core/test/model.test.ts#L470-L483

**Comment by mattpocock** [comment]
@Andarist good spot, removed

**Review by Andarist — APPROVED** [review]

**Review by davidkpiano — APPROVED** [review]

**Review by farskid — APPROVED** [review]

**Pull request merged**